src/task4feedback/simulator/rl/models/oracles.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC,abstractmethod
+
+import torch
+
+from ....types import Device, Architecture
+
+class OraclePolicy(ABC):
+
+  def get_action(self, system_state: "SystemState"):
+      NotImplementedError()
+
+
+class LoadbalancingPolicy(ABC):
+
+
+  def get_action(self, system_state: "SystemState"):
```

Comment:
Eventually for safety this should somehow:
(1) Filter for devices that can't run the current task; or 
(2) Be passed a device list that meets the current tasks requirements and makes a choice over that list.